packages/oui-dual-list/src/dual-list.html

```diff
@@ -0,0 +1,80 @@
+
+    <div class="oui-dual-list__source">
+        <div class="oui-dual-list__header">
+            <span>{{$ctrl.sourceListLabel}} <span data-ng-if="!$ctrl.sourceListLoading">({{$ctrl.sourceList.length}})</span></span>
+            <span class="oui-dual-list__header_toggle">
+                <i class="oui-icon oui-icon-chevron-up oui-dual-list__header_toggle_up"
+                    data-ng-click="$ctrl.onSourceContentClose()"
+                    aria-hidden="true"></i>
+                <i class="oui-icon oui-icon-chevron-down oui-dual-list__header_toggle_down"
+                    data-ng-click="$ctrl.onSourceContentOpen()"
+                    aria-hidden="true"></i>
+            </span>
+        </div>
+        <div class="oui-dual-list__content">
+            <div class="oui-dual-list__empty-label" data-ng-if="!$ctrl.sourceListLoading && !$ctrl.sourceList.length">
+                <p>{{$ctrl.sourceListEmptyLabel}}</p>
+            </div>
+            <div class="oui-dual-list__empty-label" data-ng-if="$ctrl.sourceListLoading">
+                <oui-spinner size="m"></oui-spinner>
+            </div>
+            <div class="oui-dual-list__content_body" data-ng-if="!$ctrl.sourceListLoading && $ctrl.sourceList.length">
+                <oui-search class="oui-dual-list__content_filter" model="$ctrl.sourceFilterKey" ng-model-options="{debounce: 500}" aria-label="search items">
+                </oui-search>
+                <div class="oui-dual-list__content_add-all" data-ng-if="$ctrl.bulkActionEnabled">
```

Comment:
Careful on your BEM naming, for `oui-dual-list__content_add-all`. I don't think `&_add-all` is a modifier.
For example, a more compliant name could be `oui-dual-list__add-all-button`, since it's an element :)

It apply too for all others classnames in this template ;)